lib/select-builder.js

```diff
@@ -92,232 +77,162 @@ class SelectBuilder extends QueryBuilder {
     return this;
   }
 
-  whereNotIn(key, conds) {
-    this.operations.get('where').push({
-      key: this.escapeKey(key),
-      cond: 'NOT IN',
-      value: conds,
-    });
+  whereNotIn(field, values) {
+    this.where(field, 'NOT IN', values);
     return this;
   }
 
-  whereAny(key, value) {
-    this.operations.get('where').push({
-      key: this.escapeKey(key),
-      cond: '= ANY',
-      value,
-    });
+  whereAny(field, values) {
+    this.where(field, '= ANY', values);
     return this;
   }
 
   whereExists(subquery) {
-    this.operations.get('where').push({
-      cond: 'EXISTS',
+    this.operations.where.push({
+      op: 'EXISTS',
       value: subquery,
     });
     return this;
   }
 
-  orderBy(field, dir = 'ASC') {
-    dir = dir.toUpperCase();
-    this.operations.get('orderBy').add({ field: this.escapeKey(field), dir });
+  orderBy(key, dir = 'ASC') {
+    const desc = dir.toUpperCase() === 'DESC';
+    const field = this.escapeKey(key);
+    this.operations.order.push({ field, desc });
     return this;
   }
 
   groupBy(...fields) {
-    const groupBy = this.operations.get('groupBy');
-    iter(fields)
-      .map(this.escapeKey)
-      .forEach(f => groupBy.add(f));
+    const keys = fields.map(this.escapeKey);
+    this.operations.group.push(...keys);
     return this;
   }
 
   limit(limit) {
     checkType(limit, 'limit', 'number');
-    this.operations.set('limit', limit);
+    this.operations.limit = limit;
     return this;
   }
 
   offset(offset) {
     checkType(offset, 'offset', 'number');
-    this.operations.set('offset', offset);
+    this.operations.offset = offset;
     return this;
   }
 
   count(field = '*') {
     if (field !== '*') field = this.escapeKey(field);
-    this.operations.get('count').push({ field });
+    this.operations.count.push({ field });
     return this;
   }
 
   avg(field) {
-    this.operations.get('avg').push({ field: this.escapeKey(field) });
+    this.operations.avg.push({ field: this.escapeKey(field) });
     return this;
   }
 
   min(field) {
-    this.operations.get('min').push({ field: this.escapeKey(field) });
+    this.operations.min.push({ field: this.escapeKey(field) });
     return this;
   }
 
   max(field) {
-    this.operations.get('max').push({ field: this.escapeKey(field) });
+    this.operations.max.push({ field: this.escapeKey(field) });
     return this;
   }
 
   sum(field) {
-    this.operations.get('sum').push({ field: this.escapeKey(field) });
+    this.operations.sum.push({ field: this.escapeKey(field) });
     return this;
   }
 
-  processSelect(query, clauses) {
-    if (clauses.size > 0) {
-      return query + ' ' + iter(clauses).reduce((acc, id) => acc + ', ' + id);
-    }
-    return query + ' *';
-  }
-
-  processOperations(query, operations, functionHandlers) {
-    for (const fn of Object.keys(functionHandlers)) {
-      const ops = operations.get(fn);
+  buildFields() {
+    const fields = [...this.operations.fields];
```

Comment:
Same as with `query` comment, we just create an array to later use `.join`, what's the point of this?

Reply:
We can rewrite it to concatenation here